api/can_api_definition.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Optional
+import pydantic
+import datetime
+
+"""
+Documented https://www.dropbox.com/scl/fi/o4bec2kz8dkcxdtabtqda/CAN-V1-API-Draft.paper?dl=0&rlkey=f7elx3zmo6tt5s7mj5nvap9a0
+
+"""
+
+class _ResourceUsageProjection(pydantic.BaseModel):
+    peakShortfall: int = pydantic.Field(
+        ..., description="Shortfall of resource needed at the peek utilization"
+    )
```

Comment:
this needs to be optional as it might be none

Reply:
alternatively it could be 0?

Reply:
i think we pass in a <null> for this so we could either make it 0 elsewhere or make it none